src/optimization/steepest_descent_method.cpp

```diff
@@ -20,282 +20,282 @@ namespace optimization
-	communicator->send(target, k52::parallel::mpi::constants::kCommonTag, max_iteration_number_);
-	communicator->send(target, k52::parallel::mpi::constants::kCommonTag, precision_);
+    communicator->send(target, k52::parallel::mpi::constants::kCommonTag, increment_of_the_argument_);
+    communicator->send(target, k52::parallel::mpi::constants::kCommonTag, max_iteration_number_);
+    communicator->send(target, k52::parallel::mpi::constants::kCommonTag, precision_);
 }
 
 void SteepestDescentMethod::Receive(boost::mpi::communicator* communicator, int source)
 {
-	communicator->recv(source, k52::parallel::mpi::constants::kCommonTag, increment_of_the_argument_);
-	communicator->recv(source, k52::parallel::mpi::constants::kCommonTag, max_iteration_number_);
-	communicator->recv(source, k52::parallel::mpi::constants::kCommonTag, precision_);
+    communicator->recv(source, k52::parallel::mpi::constants::kCommonTag, increment_of_the_argument_);
+    communicator->recv(source, k52::parallel::mpi::constants::kCommonTag, max_iteration_number_);
+    communicator->recv(source, k52::parallel::mpi::constants::kCommonTag, precision_);
 }
 #endif
 
 
 vector<double> SteepestDescentMethod::FindOptimalParameters(const vector<double>& initial_parameters)
 {
-	vector<double> parameters(initial_parameters);
-	size_t counter = 0;
-	double lambda = 1; 
+    vector<double> parameters(initial_parameters);
+    size_t counter = 0;
+    double lambda = 1; 
 
 	do
 	{
-		counter++;
-		vector<double> gradient = CalculateGradient(parameters);
-		vector<double> invert_gradiend = InvertVector(gradient);
-
-		lambda = GoldenSectionSearch(parameters, lambda, invert_gradiend);
-
-		vector<double> next_step_parameters(parameters.size());
-		for (size_t i = 0; i < next_step_parameters.size(); ++i)
-		{
-			next_step_parameters[i] = parameters[i] + lambda * invert_gradiend[i];
-		}
-
-		if (IsExitCriteriaFulfilled(next_step_parameters))
-		{
-			break;
-		}
+        counter++;
+        vector<double> gradient = CalculateGradient(parameters);
+        vector<double> invert_gradiend = InvertVector(gradient);
+
+        lambda = GoldenSectionSearch(parameters, lambda, invert_gradiend);
+
+        vector<double> next_step_parameters(parameters.size());
+        for (size_t i = 0; i < next_step_parameters.size(); ++i)
+        {
+            next_step_parameters[i] = parameters[i] + lambda * invert_gradiend[i];
+        }
+
+        if (IsExitCriteriaFulfilled(next_step_parameters))
+        {
+            break;
+        }
 		
-		parameters = next_step_parameters;
+        parameters = next_step_parameters;
 
 		
-		if (counter == max_iteration_number_)
-		{
-			std::cout << " Minimun has not found"<<std::endl;
-			break;
-		}
-	} while (true);
-
-	return parameters;
+        if (counter == max_iteration_number_)
+        {
+            std::cout << " Minimun has not found"<<std::endl;
+            break;
+        }
+    } while (true);
```

Comment:
check if max_iteration_number_ reached here instead of having infinite loop with internal break